### ISSUE TITLE
Adapt disabled vararg tests from PUC-Rio Lua 5.1 test suite in vararg.lua

### ISSUE THREAD

**Issue by Buristan** [tarantool/tarantool#5712]
This ticket is a logical continuation of gh-4473.

If run  PUC-Rio Lua 5.1 test suite by LuaJIT (HEAD at the moment: [`1672bdc`](https://github.com/tarantool/luajit/commit/1672bdc0ffbee9f32fadb6943909d6af7eb9a7b1)) (not Tarantool) it fails with the following error:
```
/home/burii/builds_workspace/luajit/gh-4473-adopt-luavela-test-suites/src/luajit: vararg.lua:6: assertion failed!
stack traceback:
        [C]: in function 'assert'
        vararg.lua:6: in function 'f'
        vararg.lua:24: in function 'dofile'
        ...adopt-luavela-test-suites/test/PUC-Lua-5.1-tests/all.lua:99: in main chunk
        [C]: at 0x558396fbcf96
```
caused by this test chunk (assuming that all previous failing test chunks non relating to this error are disabled):
```lua
function f(a, ...)
  assert(type(arg) == 'table')
  assert(type(arg.n) == 'number')
  for i=1,arg.n do assert(a[i]==arg[i]) end
  return arg.n
end
```
According to [this section](https://www.lua.org/manual/5.0/manual.html#2.5.8) in Lua 5.0 Reference Manual, Lua 5.0 collect all extra arguments in `arg` variable in the vararg functions.
> A vararg function does not adjust its argument list; instead, it collects all extra arguments into an implicit parameter, called arg. 

According to [this section](https://www.lua.org/manual/5.1/manual.html#7.1)  described changes in the language in Lua 5.1 Reference Manual:
> The vararg system changed from the pseudo-argument arg with a table with the extra arguments to the vararg expression.

**Comment by Buristan** [comment]
This test chunk can be adapted from PUC-Rio Lua 5.2 test suite like this:
```diff
diff --git a/test/PUC-Lua-5.1-tests/vararg.lua b/test/PUC-Lua-5.1-tests/vararg.lua
index e2de1ad..27d7f7b 100644
--- a/test/PUC-Lua-5.1-tests/vararg.lua
+++ b/test/PUC-Lua-5.1-tests/vararg.lua
@@ -2,9 +2,13 @@ print('testing vararg')
 
 _G.arg = nil
 
+-- Lua 5.1 interprets ... in the vararg functions like additional
+-- first argument unlike LuaJIT does. All extra arguments is set
+-- into `arg` variable. This extension is from Lua 5.2.
+-- See also https://github.com/tarantool/tarantool/issues/5712.
+-- Test is adapted from PUC-Rio Lua 5.2 test suite.
 function f(a, ...)
-  assert(type(arg) == 'table')
-  assert(type(arg.n) == 'number')
+  local arg = {n = select('#', ...), ...}
   for i=1,arg.n do assert(a[i]==arg[i]) end
   return arg.n
 end
@@ -17,18 +21,15 @@ function c12 (...)
   return res, 2
 end
 
-function vararg (...) return arg end
+-- Function is adapted from PUC-Rio Lua 5.2 test suite.
+-- See the comment above.
+function vararg (...) return {n = select('#', ...), ...} end
 
 local call = function (f, args) return f(unpack(args, 1, args.n)) end
 
--- Lua 5.1 interprets ... in the vararg functions like additional
--- first argument unlike LuaJIT does. All extra arguments is set
--- into `arg` variable. This extension is from Lua 5.2.
--- See also https://github.com/tarantool/tarantool/issues/5712.
--- Test is disabled for LuaJIT.
--- assert(f() == 0)
--- assert(f({1,2,3}, 1, 2, 3) == 3)
--- assert(f({"alo", nil, 45, f, nil}, "alo", nil, 45, f, nil) == 5)
+assert(f() == 0)
+assert(f({1,2,3}, 1, 2, 3) == 3)
+assert(f({"alo", nil, 45, f, nil}, "alo", nil, 45, f, nil) == 5)
 
 assert(c12(1,2)==55)
 a,b = assert(call(c12, {1,2}))
@@ -40,18 +41,15 @@ assert(not a)
 assert(c12(1,2,3) == false)
 local a = vararg(call(next, {_G,nil;n=2}))
 local b,c = next(_G)
--- Test is disabled for LuaJIT for now. See the comment above.
--- assert(a[1] == b and a[2] == c and a.n == 2)
+assert(a[1] == b and a[2] == c and a.n == 2)
 a = vararg(call(call, {c12, {1,2}}))
--- Test is disabled for LuaJIT for now. See the comment above.
--- assert(a.n == 2 and a[1] == 55 and a[2] == 2)
+assert(a.n == 2 and a[1] == 55 and a[2] == 2)
 a = call(print, {'+'})
 assert(a == nil)
 
 local t = {1, 10}
-function t:f (...) return self[arg[1]]+arg.n end
--- Test is disabled for LuaJIT for now. See the comment above.
--- assert(t:f(1,4) == 3 and t:f(2) == 11)
+function t:f (...) local arg = {...}; return self[...]+#arg end
+assert(t:f(1,4) == 3 and t:f(2) == 11)
 print('+')
 
 lim = 20

```

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@5c0d2e7 . The submodule was updated in tarantool in 2.8.0-249-g8a09e18ce (8a09e18), 2.7.1-210-g5f4298b6f (5f4298b), 2.6.2-203-g8343a8129 (8343a81), 1.10.9-119-g7a4c7d0b0 (7a4c7d0).
